config/default.yml

```
@@ -1568,6 +1568,11 @@ Lint/CircularArgumentReference:
   Enabled: true
   VersionAdded: '0.33'
 
+Lint/ConstantBase:
```

It's a style cop and I think adding "Redundant" to the name makes it consistent.

```suggestion
Style/RedundantConstantBase:
```

---

This Cop is very similar to `Lint/ConstantResolution` in the sense that it decides whether or not to add `::` to a constant.

Can you elaborate a bit more on what the difference is between Lint and Style here? I'm not quite sure I understand it yet. I may have missed some documents about each department, so if you know of any, I'd be glad to know 🙏

---

`Lint/ConstantResolution` cop is intended to prevent bugs as wrote in its description. OTOH, this cop looks for consistency of style. The purpose of the cop is different.

I wrote it on the slide I talked about at past RubyKaigi for the difference is between Lint and Style. Here's what I can quickly recall.
https://speakerdeck.com/koic/improve-ruby-coding-styles-and-lint?slide=55

---

As a side note, it's worth remembering that Lint is the special department for Severity.
https://github.com/rubocop/rubocop/issues/10080#issuecomment-1006148714

---

Thank you so much, and that's a good resource :+1:

I reconsidered that it should be in the `Style` department. It seems unlikely that the presence or absence of this `::` will cause any grammatical problems or bugs, so we can say that it is a matter of preference whether this Cop should be followed or not. In that sense, as you say, this should not be in the `Lint` department.